atest/robot/variables/variable_types.robot

```diff
@@ -0,0 +1,164 @@
+*** Settings ***
+Suite Setup       Run Tests    ${EMPTY}    variables/variable_types.robot
+Resource          atest_resource.robot
+
+*** Test Cases ***
+Variable section
+    Check Test Case    ${TESTNAME}
+
+Variable section: list
+    Check Test Case    ${TESTNAME}
+
+Variable section: dictionary
+    Check Test Case    ${TESTNAME}
+
+Variable section: with invalid values or types
+    Check Test Case    ${TESTNAME}
+
+Variable section: parings errors
+    Error In File
+    ...    2    variables/variable_types.robot
+    ...    17    Setting variable '\${BAD_TYPE: hahaa}' failed: Unrecognized type 'hahaa'.
+    Error In File
+    ...    3    variables/variable_types.robot    19
+    ...    Setting variable '\@{BAD_LIST_TYPE: xxxxx}' failed: Unrecognized type 'xxxxx'.
+    Error In File
+    ...    4    variables/variable_types.robot    21
+    ...    Setting variable '\&{BAD_DICT_TYPE: aa=bb}' failed: Unrecognized type 'aa'.
+    Error In File
+    ...    5    variables/variable_types.robot    22
+    ...    Setting variable '\&{INVALID_DICT_TYPE1: int=list[int}' failed: Parsing type
+    ...    'dict[int, list[int]' failed: Error at end: Closing ']' missing.
```

Comment:
In cases like this, splitting after the `:` would be a bit nice. No need to split after every `:`, though. This could look like below, but two last rows could also be merged.
```
...    Setting variable '\&{INVALID_DICT_TYPE1: int=list[int}' failed:
...    Parsing type 'dict[int, list[int]' failed:
...    Error at end: Closing ']' missing.
```

Reply:
Done